core-blocks/file/index.js

```diff
@@ -96,9 +97,10 @@ export const settings = {
 				type: 'block',
 				blocks: [ 'core/audio' ],
 				transform: ( attributes ) => {
+					const captionProcessed = getTextElements( attributes.caption );
```

Comment:
`caption` is a `children` value. It's not expected to be strictly compatible with `element`. The implementation of `getTextElements` in `@wordpress/element` and its usage with a `children` value contradicts this.

See #7934